File: SDKs/vmware.local/vrni/1.0.0/python/src/sdk/models/operations/getdellswitch.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class GetDellSwitchPathParams:
    id: str = field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetDellSwitchSecurity:
    api_key_auth: shared.SchemeAPIKeyAuth = field(metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    

@dataclass
class GetDellSwitchRequest:
    path_params: GetDellSwitchPathParams = field()
    security: GetDellSwitchSecurity = field()
    

@dataclass
class GetDellSwitchResponse:
    content_type: str = field()
    status_code: int = field()
    dell_switch_data_source: Optional[shared.DellSwitchDataSource] = field(default=None)
    

File: SDKs/vmware.local/vrni/1.0.0/typescript/src/sdk/models/operations/getdellswitch.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetDellSwitchPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class GetDellSwitchSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth: shared.SchemeApiKeyAuth;
}


export class GetDellSwitchRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetDellSwitchPathParams;

  @SpeakeasyMetadata()
  security: GetDellSwitchSecurity;
}


export class GetDellSwitchResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  dellSwitchDataSource?: shared.DellSwitchDataSource;

  @SpeakeasyMetadata()
  statusCode: number;
}
